### PR TITLE
Added User class and sign up ability

### PR DESCRIPTION
**Pull request by abenzzine10**
Added the User class with the following functionalities:

- Create a new user
- Get a user by username
- Check that a username is unique
- User authentication

Also, added the sign-up ability to the menu.

### CODE REVIEW

File: src/Menu.java [hunk 1]
```diff
@@ -72,25 +72,66 @@ public static void login() {
...
-			return true;
-		} else {
-			return false;
+		return User.authentication(username, password);
+	}
+	
+	/**
+	 * User can Enter a valid username and password to create a new account
+	 */
+	public static void signup() {
```
Comment (andrewmitra) [inline src/Menu.java]: Can you break this up into smaller functions? It's just a little long.

Comment (andrewmitra) [reply src/Menu.java]: Also while you are fixing this, could you remove the TransactionTest.file and the test I made for the isValidUsername andPassword?

**Comment by abenzzine10** [comment]
- Split the Sign Up method into three parts
- Deleted the TransactionTest file
- Deleted the od isUsernameAndPasswordValid

**Review by andrewmitra — APPROVED** [review]

**Comment by andrewmitra** [comment]
approved

**Pull request merged**